mpt_trie/src/utils.rs

```diff
@@ -4,45 +4,29 @@ use std::{
     borrow::Borrow,
     fmt::{self, Display},
     ops::BitAnd,
-    sync::Arc,
 };
 
 use ethereum_types::H256;
 use num_traits::PrimInt;
 
 use crate::{
     nibbles::{Nibble, Nibbles, NibblesIntern},
-    partial_trie::{Node, PartialTrie},
+    partial_trie::Node,
     trie_ops::TrieOpResult,
 };
 
-#[derive(Clone, Debug, Eq, Hash, PartialEq)]
 /// Simplified trie node type to make logging cleaner.
-pub enum TrieNodeType {
-    /// Empty node.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum NodeKind {
     Empty,
-
-    /// Hash node.
     Hash,
-
-    /// Branch node.
     Branch,
-
-    /// Extension node.
     Extension,
-
-    /// Leaf node.
     Leaf,
 }
 
-impl<N: PartialTrie> From<&Arc<Box<N>>> for TrieNodeType {
-    fn from(value: &Arc<Box<N>>) -> Self {
-        (&****value).into()
-    }
-}
-
-impl<N: PartialTrie> From<&Node<N>> for TrieNodeType {
-    fn from(node: &Node<N>) -> Self {
+impl NodeKind {
+    pub fn of(node: &Node) -> Self {
```

Comment:
nit: missing doc, `of` isn't really self-explanatory

Reply:
Ah, I was going for a https://doc.rust-lang.org/std/any/struct.TypeId.html#method.of -style API

Reply:
Ah ok, maybe it's just me who wasn't aware of this style then :)

Reply:
This is basically also https://docs.rs/enum-kinds/latest/enum_kinds/, so it's a bit of a pattern. I'll add some light docs :)